cmd/delete/cluster.go

```diff
@@ -61,22 +67,40 @@ func NewClusterCmd() *cobra.Command {
 				return ErrForceFlagNotSet
 			}
 
-			execx.Debug = debug
-
 			// Init paths.
 			homeDir, err := os.UserHomeDir()
 			if err != nil {
 				return fmt.Errorf("error while getting user home directory: %w", err)
 			}
 
-			minimalConf, err := yamlx.FromFileV3[config.Furyctl](furyctlPath)
+			// Init first half of collaborators.
+			client := netx.NewGoGetterClient()
+			executor := execx.NewStdExecutor()
+			distrodl := distribution.NewDownloader(client, debug)
+
+			execx.Debug = debug
+
+			// Download the distribution.
+			logrus.Info("Downloading distribution...")
+			res, err := distrodl.Download(version, distroLocation, furyctlPath)
```

Comment:
Is this "smart" and avoid downloading the distro if it's already present on the filesystem?

Reply:
nope every time we download the distro to a temp folder

Reply:
@ralgozino thoughts on this?

Reply:
IMO as a first approach it is OK to download everytime and start from a clean state. The distro download is not a resource nor time intensive operation.

Once we have a clear overall picture we can start to optimize this kind of things.

Reply:
ok